### PR TITLE
Ensure a member exists before calling `isLockedOut`

### PR DESCRIPTION
**Pull request by jakedaleweb**
Keep seeing `PHP Fatal error: Call to a member function isLockedOut() on null` in logs, this should fix that.

### CODE REVIEW

File: security/MemberAuthenticator.php [hunk 1]
```diff
@@ -51,7 +51,9 @@ protected static function authenticate_member($data, $form, &$success) {
 		if($asDefaultAdmin) {
 			// If logging is as default admin, ensure record is setup correctly
 			$member = Member::default_admin();
```
Comment (tractorcow) [inline security/MemberAuthenticator.php]: I think the fix is in the wrong place. Member::default_admin() should NOT be returning null if a default admin is specified. that method is meant to be guaranteed to return an instance. Can you please investigate why this is not happening?

File: security/MemberAuthenticator.php [hunk 2]
```diff
@@ -51,7 +51,9 @@ protected static function authenticate_member($data, $form, &$success) {
...
-			$success = !$member->isLockedOut() && Security::check_default_admin($email, $data['Password']);
+			if($member && !is_null($member)) {
```
Comment (tractorcow) [inline security/MemberAuthenticator.php]: The right hand condition is redundant here, in any case.

**Comment by dhensby** [comment]
I think this is just an order of comparison issue.

**Comment by dhensby** [comment]
related #6944

**Pull request closed without merging**